src/main/scala/jio/package.scala

```diff
@@ -74,7 +64,37 @@ package object jio extends JioFiles with DecorateAsScala with DecorateAsJava {
     }
   }
 
-  trait Pathish[Rep] {
+  trait Metadataish {
+    def exists: Boolean
+    def path: Path
+    def allBytes: Array[Byte]
+    def nodeType: fs.NodeType
+    def size: Long
+    def permissions: PosixFilePermissions
+    def atime: Long
+    def mtime: Long
+
+    def name : String    = path.getFileName.to_s
+    def blockCount: Long = (size + blockSize - 1) / blockSize
+    def blockSize: Long  = 512 // FIXME
+  }
+
+  // this could have been an apply method in an object allowing the use of Metadata instead of Metadataish
+  // I however could not get myself to repeat all those arguments
```

Comment:
There should be fewer arguments though. I would (and have - still so much already written code to chase down) break these down into a `Perms(uid, gid, rwxrwxrwx)` and a `Times(atime, mtime, ctime, btime)` at least. As you probably know, the actual bytes don't belong attached to the metadata. Nor does the path really, and the size doesn't either. Unix has all manner of 1970s-style sickness (like using the size field for whatever came to mind in the non-file file variants) but we have sum types available. Only a file-file has a quality which can sensibly be called size.

Not sure about attaching the name to the metadata. Really it's a property of the containing directory - the name is the key, the metadata is the value. The characteristic data of the key types, assuming classical semantics, is
```
File: Array[Byte]
Dir: Map[Name, Inum]
Link: Path
```
Where `Name` is a String without slashes. In this presentation the map is from names to Inums, but an Inum is a key which uniquely identifies the metadata.